### PR TITLE
feature/SOF-3783

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: express/parsers/apps/espresso/formats/txt.py [hunk 1]
```diff
@@ -721,3 +721,38 @@ def charge_density_profile(self, text):
...
+            Sigma_ex val (eV)  -16.04  -15.82  -15.82  -15.82   -3.54   -3.54   -3.54   -4.03
+            QP renorm            1.00    1.00    1.00    1.00    1.00    1.00    1.00    1.00
+
+        Returns:
+            list[]
+        """
+        kpoints = self._general_output_parser(text, **settings.REGEX["gw_kpoint"])
+        eigenvalues = self._general_output_parser(text, **settings.REGEX["gw_eigenvalues"])
+        eigenvalues = [map(float, re.sub(' +', ' ', e).strip(" ").split(" ")) for e in eigenvalues]
+        return [
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/formats/txt.py]: We need to explain the logic for setting the weights and occupations below, perhaps in inline comments

File: express/parsers/apps/espresso/parser.py [hunk 2]
```diff
@@ -51,14 +64,34 @@ def nspins(self):
...
+        NOTE: DO NOT READ THE WHOLE FILE INTO MEMORY.
+
+        Returns:
+             bool
+        """
+        with open(self.stdout_file, "r") as f:
+            for index, line in enumerate(f):
+                if index > 50: return False
+                if "SternheimerGW" in line:
+                    return True
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/parser.py]: We need to explain the logic in docstring - what do you actually do/assume here about the file?

File: express/parsers/apps/espresso/parser.py [hunk 3]
```diff
@@ -22,7 +22,20 @@ def __init__(self, *args, **kwargs):
         self.work_dir = self.kwargs["work_dir"]
         self.stdout_file = self.kwargs["stdout_file"]
         self.txt_parser = EspressoTXTParser(self.work_dir)
-        self.xml_parser = EspressoXMLParser(find_file(settings.XML_DATA_FILE, self.work_dir))
+        self.xml_parser = EspressoXMLParser(self.find_xml_file())
+
+    def find_xml_file(self):
+        """
+        Finds XML file. GW XML file is returned if this a GW calculation.
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/parser.py]: We should be careful with naming here b/c QE has multiple GW implementations (eg. GWW). We should call GW "SteinheimerGW" everywhere instead

File: express/parsers/apps/espresso/settings.py [hunk 4]
```diff
@@ -125,7 +125,17 @@
...
         "match_groups": [1, 2]
     },
-
+    "gw_kpoint": {
+        "regex": r"^\s+GWKpoint cryst:\s+({0})\s+({0})\s+({0})".format(DOUBLE_REGEX),
+        "occurrences": 0,
+        "output_type": "float",
+        "match_groups": [1, 2, 3]
+    },
+    "gw_eigenvalues": {
```
Comment (timurbazhirov) [inline express/parsers/apps/espresso/settings.py]: steinheimer_gw_kpoint
steinheimer_gw_eigenvalues

File: express/parsers/apps/espresso/formats/txt.py (hunk 1 above)
Comment (mohammadidinani) [reply express/parsers/apps/espresso/formats/txt.py]: Done.

**Pull request merged**